### PR TITLE
Fixes #10802: Fix error that appears in migrations related to users.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: config/initializers/foreman.rb [hunk 1]
```diff
@@ -28,10 +28,12 @@
 Dashboard::Loader.load
 
 # clear our users topbar cache
-begin
-  User.unscoped.pluck(:id).each do |id|
-    Rails.cache.delete("views/tabs_and_title_records-#{id}")
+if defined?(User)
```
Comment (dLobatog) [inline config/initializers/foreman.rb:33]: I'd say `if Foreman.in_rake?('db:migrate')` to make it more clear.

**Comment by dLobatog** [comment]
@ehelms Minor comment, I was able to reproduce on PG too consistently. Thanks!

File: config/initializers/foreman.rb (hunk 1 above)
Comment (ehelms) [inline config/initializers/foreman.rb:33]: That feels limiting given we are trying to access a model in an initializer and there are states of the application where that model or its tables may or may not exist yet.

File: config/initializers/foreman.rb [hunk 2]
```diff
@@ -28,10 +28,12 @@
...
+  begin
+    User.unscoped.pluck(:id).each do |id|
+      Rails.cache.delete("views/tabs_and_title_records-#{id}")
+    end
+  rescue => e
```
Comment (domcleal) [inline config/initializers/foreman.rb]: Can we remove the rescue now?

File: config/initializers/foreman.rb (hunk 1 above)
Comment (domcleal) [inline config/initializers/foreman.rb:33]: Yeah, I'm fine with leaving it with a `defined?` call only, but I'd suggest putting a comment next to it instead to make the intention clear.

**Comment by ehelms** [comment]
Removed the rescue and added a comment.

**Comment by domcleal** [comment]
Merged as 7faa996, thanks @ehelms.

**Pull request closed without merging**